src/library_pthread.js

```diff
@@ -790,7 +790,16 @@ var LibraryPThread = {
     if (canceled == 2) throw 'Canceled!';
   },
 
-  {{{ USE_LSAN ? 'emscripten_builtin_' : '' }}}pthread_join__deps: ['_cleanup_thread', '_pthread_testcancel_js', 'emscripten_main_thread_process_queued_calls', 'emscripten_futex_wait'],
+  emscripten_block_on_main_thread: function() {
```

Comment:
Maybe a nit pick, but I don't really like the name of this function. To me it sounds like this blocks on the main thread, when it actually either aborts of does nothing.

Reply:
I agree `emscripten_block_on_main_thread` is not the best, but I couldn't find a better one - ideas?

Reply:
I'm not sure. The only thing I can think of would be emscripten_block_on_main_thread_or_die (in the same vain as functions like malloc_or_die), but I'm not sure if that's actually any better. Either way it's not that big of a deal in my opinion.